pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -385,14 +394,18 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule, push *model.PushContex
 			},
 		},
 		Providers: providers,
-	}
+	}, nil
 }
 
 // buildLocalJwks builds local Jwks by fetching the Jwt Public Key from the URL passed if it is empty.
-func buildLocalJwks(jwksURI, jwtPubKey string) *envoy_jwt.JwtProvider_LocalJwks {
+func buildLocalJwks(jwksURI, jwtPubKey string, jwtKeyResolver *model.JwksResolver) (*envoy_jwt.JwtProvider_LocalJwks, error) {
 	if jwtPubKey == "" {
 		var err error
-		jwtPubKey, err = model.GetJwtKeyResolver().GetPublicKey(jwksURI)
+
+		if jwtKeyResolver == nil {
+			return nil, fmt.Errorf("JWT key resolver not found")
+		}
+		jwtPubKey, err = jwtKeyResolver.GetPublicKey(jwksURI)
 		if err != nil {
 			log.Errorf("Failed to fetch jwt public key from %q: %s", jwksURI, err)
```

Comment:
When err != nil in (jwtPubKey, err = jwtKeyResolver.GetPublicKey(jwksURI)), why does the function return a fake key rather than returning an error?

Reply:
emm... This is a temporary solution for https://github.com/istio/istio/issues/24629 I think.

Reply:
cc @xulingqing